test/test_rlhf.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import zipfile
+from copy import deepcopy
+from pathlib import Path
+
+import datasets
+import numpy as np
+import pytest
+
+from _utils_internal import get_default_devices
+from tensordict import is_tensor_collection, MemmapTensor, TensorDict, TensorDictBase
+from torchrl.data.rlhf import TensorDictTokenizer
+from torchrl.data.rlhf.dataset import (
+    create_or_load_dataset,
+    dataset_to_tensordict,
+    get_dataloader,
+    load_dataset,
+    tokenize,
+)
+from torchrl.data.rlhf.prompt import PromptData, PromptTensorDictTokenizer
+from torchrl.data.rlhf.reward import PairwiseDataset, pre_tokenization_hook
+from transformers import AutoTokenizer
+
+HERE = Path(__file__).parent
+
+
+@pytest.fixture
+def tmpdir1(tmp_path_factory):
+    yield tmp_path_factory.mktemp("tmpdir1")
+
+
+@pytest.fixture
+def tmpdir2(tmp_path_factory):
+    yield tmp_path_factory.mktemp("tmpdir2")
+
+
+@pytest.mark.parametrize("max_length", [12, 550])
+@pytest.mark.parametrize(
+    "dataset_path,make_process_fn,pre_tokenization_hook",
+    [
+        (
+            f"{HERE}/assets/openai_summarize_comparisons.zip",
+            TensorDictTokenizer,
+            pre_tokenization_hook,
+        ),
+        (
+            f"{HERE}/assets/openai_summarize_tldr.zip",
+            PromptTensorDictTokenizer,
+            None,
+        ),
+    ],
+)
+def test_create_or_load_dataset(
+    tmpdir1,
+    tmpdir2,
+    max_length,
+    dataset_path,
+    make_process_fn,
+    pre_tokenization_hook,
+    mocker,
+):
+    # test caching of the values
+    lmemmap_save = deepcopy(TensorDict.load_memmap)
+    mocked_hello = mocker.patch("tensordict.TensorDict.load_memmap")
+    mocked_hello.side_effect = lmemmap_save
+    with zipfile.ZipFile(dataset_path, "r") as zip_ref:
```

Comment:
Instead of doing this in every test that loads the data, perhaps we should write a `conftest.py` file that extracts on setup, then cleans up on completion?

Reply:
Good point yeah
But then how do we tell conftest to do it only when the test_rlhf.py script is executed?

Reply:
> But then how do we tell conftest to do it only when the test_rlhf.py script is executed?

How about instead of `conftest.py` we add a fixture that has session scope and yields a directory which is the location of the extracted data?

Reply:
I think fixtures are executed in each test, otherwise the same temporary directories would be reused multiple times

Reply:
Yes, that's the default, but you can change that behaviour

```python
@pytest.fixture(scope="session")
def data_dir():
    # unzip
    # yield path
```

Reply:
Oh now i remember why i did not do that
we have 2 mini datasets but you can't pass a fixture to parametrize, so I couldn't write 2 different fixtures for the tests
The "ugly" solution would be

```
@pytest.mark.parametrize("dataset", ["comp", "prompt"])
def test_stuff(dataset, comp_fixture, prompt_fixture):
    if dataset == "comp":
        dataset = comp_fixture
    else:
        dataset = prompt_fixture
```